gnc/sub8_perception/nodes/buoy_2d.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+import matplotlib.pyplot as plt
+
+
+import cv2
+import numpy as np
+import sys
+import rospy
+from sub8_vision_tools import threshold_tools
+from sub8_msgs.srv import VisionRequest2DResponse, VisionRequest2D
+import sub8_ros_tools
+# from geometry_msgs.msg import Pose2D
+
+
+class BuoyFinder:
+    def __init__(self):
+        self.last_image = None
+        self.last_draw_image = None
+        self.last_poop_image = None
+
+        self.pose_service = rospy.Service("vision/buoy/2D", VisionRequest2D, self.request_buoy)
+        self.image_sub = sub8_ros_tools.Image_Subscriber('/stereo/left/image_raw', self.image_cb)
+        self.image_pub = sub8_ros_tools.Image_Publisher("/vision/buoy_2d/target_info")
+
+        # self.poop_pub = sub8_ros_tools.Image_Publisher("/vision/channel_marker/thresh")
+
+        # Occasional status publisher
+        self.timer = rospy.Timer(rospy.Duration(0.1), self.publish_target_info)
+
+        self.buoys = {
+            'green': '/color/buoy/green',
+            'red': '/color/buoy/red',
+            'yellow': '/color/buoy/yellow',
+        }
+
+    def request_buoy(self, srv):
+        buoy_location = self.find_single_buoy(np.copy(self.last_image), srv.target_name)
+
+        # Fill in
+        resp = VisionRequest2DResponse()
+        return resp
+
+    def publish_target_info(self, *args):
+        if self.last_image is None:
+            return
+
+        self.find_buoys(np.copy(self.last_image))
+        if self.last_draw_image is not None:
+            self.image_pub.publish(self.last_draw_image)
+
+    def image_cb(self, image):
+        '''Hang on to last image'''
+        self.last_image = image
+
+    def ncc(self, image, mean_thresh, scale=15):
+        '''Compute normalized cross correlation w.r.t a shadowed pillbox fcn
+
+        TODO:
```

Comment:
Nice catch!

- Jake

> On Apr 28, 2016, at 3:00 AM, mattlangford <notifications@github.com> wrote:
> 
> In gnc/sub8_perception/nodes/buoy_2d.py:
> 
> > +    def publish_target_info(self, *args):
> > +        if self.last_image is None:
> > +            return
> > +
> > +        self.find_buoys(np.copy(self.last_image))
> > +        if self.last_draw_image is not None:
> > +            self.image_pub.publish(self.last_draw_image)
> > +
> > +    def image_cb(self, image):
> > +        '''Hang on to last image'''
> > +        self.last_image = image
> > +
> > +    def ncc(self, image, mean_thresh, scale=15):
> > +        '''Compute normalized cross correlation w.r.t a shadowed pillbox fcn
> > +
> > +        TODO:
> Could use Cache decorator here instead of recomputing.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>